### DISCUSSION TITLE
tol47wifi thermostat unavailable after some time

### DISCUSSION THREAD

**Discussion by vasyna** [make-all/tuya-local#617] [Q&A]
Often the thermostat becomes unavailable. I manage by restarting the integration. Is there any way to make the integration try to restart itself? And what can I do to make the connection more stable? The device is not disconnected from the Wi-Fi network, ip is assigned static.
<img width="399" alt="image" src="https://user-images.githubusercontent.com/4531827/232206896-0d2f1279-dbca-4c31-b449-ad161ee65e81.png">
Here I have not yet created an automation for restarting. As you can see, the device becomes unavailable with enviable regularity. thanks.

**Comment by make-all** [comment]
Looking at the timing: morning, midday and evening, microwave oven perhaps?

**Reply by vasyna** [reply]
> Looking at the timing: morning, midday and evening, microwave oven perhaps?

No, it is far away and there are no problems with other WiFi devices.

**Reply by vasyna** [reply]
> Looking at the timing: morning, midday and evening, microwave oven perhaps?

Most likely, the device simply has not yet taken root in the house)))
In the last 48 hours, only 2 thermostats became unavailable, and then they were immediately restarted by the trigger. while I continue to monitor the work. Thank you
